pipeline_dp/accumulator.py

```diff
@@ -222,11 +225,26 @@ def add_accumulator(
         self.add_value(accumulator._vec_sum)
         return self
 
+    def _clip(self, vec_sum: np.ndarray, max_norm: float, norm_kind="linf"):
+        if norm_kind == "linf":
+            return np.clip(vec_sum, -max_norm, max_norm)
+        elif norm_kind in ["l0", "l1", "l2"]:
+            norm_kind = int(norm_kind[-1])
+            vec_norm = np.linalg.norm(vec_sum, ord=norm_kind)
+            mul_coef = min(1, max_norm/vec_norm)
+            return vec_sum * mul_coef
+
     def compute_metrics(self):
-        # TODO - add DP anonymization
         if self._vec_sum is None:
             raise IndexError("No data provided for metrics computation.")
-        return self._vec_sum
+        if self._params is None:
+            return self._vec_sum
+        vec_sum = self._clip(self._vec_sum, self._params.max_norm, self._params.clip_norm_kind)
+        vec_sum = np.array([
```

Comment:
let's make a function add_noise_to_vector in dp_computations, which adds noise to the vector. And let's create AddNoiseToVectorParams class

Reply:
Done.